src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -47,6 +49,40 @@ public Optional<String> getDerivedTableName(UUID entityTypeId) {
       .fetchOptional(derivedTableNameField);
   }
 
+  public List<EntityTypeColumn> fetchNamesForSingleCheckbox(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getDerivedTableName(entityTypeId).get();
+    List<EntityTypeColumn> entityColumnsList = new ArrayList<>();
+
+    String requiredFieldName = "jsonb" + "->>'name'";
+    String abc =  "jsonb" + "->>'refId'";
+    String whereClauseName = "jsonb" + "->>'type'";
+    String customFieldType = "SINGLE_CHECKBOX";
+    ValueWithLabel valueWithLabelTrue = new ValueWithLabel().label("True").value("true");
+    ValueWithLabel valueWithLabelFalse = new ValueWithLabel().label("False").value("false");
+
+    Result<Record2<Object, Object>> result = jooqContext
+      .select(field(requiredFieldName), field(abc))
+      .from(sourceViewName)
+      .where(field(whereClauseName).eq(customFieldType))
+      .fetch();
+    result.stream()
+      .map(record -> {
+        Object value = record.get(0);
+        EntityTypeColumn entityTypeColumn = new EntityTypeColumn();
+        entityTypeColumn.name((value instanceof List) ?
+          ((List<?>) value).stream().map(item -> item != null ? item.toString() : null).collect(Collectors.joining(", "))
```

Comment:
If `value` _can_ be a `List`, then I'd use pattern matching to remove the cast here

```suggestion
        entityTypeColumn.name((value instanceof List valueList) ?
          valueList.stream().map(item -> item != null ? item.toString() : null).collect(Collectors.joining(", "))
```